x-pack/plugins/security_solution/public/detection_engine/rule_management_ui/components/rules_table/rules_table/use_initialize_rules_table_saved_state.ts

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { useEffect } from 'react';
+import { useGetInitialUrlParamValue } from '../../../../../common/utils/global_query_string/helpers';
+import { RULES_TABLE_MAX_PAGE_SIZE } from '../../../../../../common/constants';
+import { useKibana } from '../../../../../common/lib/kibana';
+import { URL_PARAM_KEY } from '../../../../../common/hooks/use_url_state';
+import { RULES_TABLE_STATE_STORAGE_KEY } from '../constants';
+import { useRulesTableContext } from './rules_table_context';
+import type { RulesTableSavedState } from './rules_table_saved_state';
+import { DEFAULT_FILTER_OPTIONS, DEFAULT_SORTING_OPTIONS } from './rules_table_defaults';
+
+export function useInitializeRulesTableSavedState(): void {
+  const getUrlParam = useGetInitialUrlParamValue<RulesTableSavedState>(URL_PARAM_KEY.rulesTable);
+  const { actions } = useRulesTableContext();
+  const {
+    services: { sessionStorage },
+  } = useKibana();
+
+  useEffect(() => {
+    const { decodedParam: urlState } = getUrlParam();
```

Comment:
`getUrlParam` shouldn't fail but it depends on the `query-string` external package. It parses a query string if it can or returns an empty object. There are no expected exceptions here. If Rison decoding fails then `decodedParam` is just `null`.

In my opinion it's responsibility of the `useGetInitialUrlParamValue` hook to make sure there are no errors thrown but it looks safe from the code and I'd like to avoid over engineering here.

> What happens when the query string doesn't match the RulesTableSavedState interface?

If it's parsable so it's returned and may not match the TS type. Though the type is optional which means that every field can be missed. This way it's absolutely safe to rename, remove or add new fields. It will lead to just disabling part of the functionality when an outdated link is used.

@banderror if you have some specific suggestions regarding error handling in `useGetInitialUrlParamValue` it'd be nice to hear.